controller/src/main/java/io/pravega/controller/store/ArtifactStore.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.store;
+
+import io.pravega.common.concurrent.Futures;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+
+public interface ArtifactStore {
```

Comment:
Let us evaluate whether it makes sense to create this higher level abstraction right now. 
we can create this abstraction when we think tables will also require a waiting request processor mechanism for fairness. 

the whole waiting request processor business is a bit of a patchwork done for streams because we wanted fairness in execution of work for commit and other workflows.. primarily because commit can be overwhelmingly more frequent than other workflows which can end up waiting unnecessarily.. 
maybe we dont need such construct for tables ever.. at least not now.. 
so creating this abstraction is unnecessary at this point and making these methods available and implemented in table store unnecessarily.

Reply:
Removed this.